### PR TITLE
Replace Http Server + Add WebSockets

### PR DESCRIPTION
**Pull request by pythongosssss**
As discussed, implemented WebSockets to replace the current polling.
I was going to move the classes out to separate files, but for ease of reviewing i've left them in main but it might be worth you splitting them out

### CODE REVIEW

**Comment by comfyanonymous** [comment]
Seems to work and the good news is that I noticed that aiohttp seems to already get pulled in by one of the dependencies.

Small issue that I found is that I need to do CTRL-C twice to kill it now?

File: webshit/index.html [hunk 1]
```diff
@@ -480,14 +480,56 @@
...
 
-setInterval(function(){
-  fetch('/prompt')
-    .then(response => response.json())
-    .then(data => {
-        document.getElementById("queuesize").innerHTML = "Queue size: " + data.exec_info.queue_remaining + "";
-    }).catch((response) => {document.getElementById("queuesize").innerHTML = "Queue size: ERR"});
-}, 500);
+        let opened = false;
+        ws = new WebSocket(`ws://${location.host}/ws`);
```
Comment (comfyanonymous) [inline webshit/index.html]: You should detect if it's wss or ws in case someone tries to use https.

File: main.py [hunk 2]
```diff
@@ -336,35 +352,79 @@ def delete_queue_item(self, function):
...
+        self.app = web.Application()
+        routes = web.RouteTableDef()
+
+        @routes.get('/ws')
+        async def websocket_handler(request):
+            return await self.socket_handler.process(request)
+        
+        @routes.get("/")
+        async def get_root(request):
+            return aiohttp.web.HTTPFound('/index.html')
```
Comment (comfyanonymous) [inline main.py]: Can you make it return index.html instead of a redirect?

**Comment by pythongosssss** [comment]
Investigating the CTRL+C issue, I don't have it on Windows, trying to get it running via WSL

File: main.py [hunk 3]
```diff
@@ -471,6 +532,9 @@ def run(prompt_queue, address='', port=8188):
     except:
         pass
 
-    run(q, address=address, port=port)
+    try:
+        loop.run_until_complete(run(q, socket_handler, address=address, port=port))
+    except KeyboardInterrupt:
```
Comment (comfyanonymous) [inline main.py]: My ctrl-c issue is fixed if I remove this try/except.

Comment (pythongosssss) [reply main.py]: I've removed it

**Pull request merged**